frameworks/PHP/symfony/src/Controller/BenchController.php

```diff
@@ -2,108 +2,134 @@
+        if ($queries === 1 && !$request->query->has('queries')) {
             $worlds = $worlds[0];
         }
 
         return new JsonResponse($worlds);
     }
 
-    public function dbRawAction(Request $request)
+    /**
+     * @Route("/db-raw")
+     * @throws Exception
+     */
+    public function dbRawAction(Request $request): JsonResponse
     {
         $queries = $request->query->getInt('queries', 1);
         $queries = min(max($queries, 1), 500);
 
         // possibility for enhancement is the use of SplFixedArray -> http://php.net/manual/de/class.splfixedarray.php
-        $worlds = array();
+        $worlds = [];
         $conn = $this->get('database_connection');
 
         for ($i = 0; $i < $queries; ++$i) {
-            $worlds[] = $conn->fetchAssoc('SELECT * FROM world WHERE id = ?', array(mt_rand(1, 10000)));
+            $worlds[] = $conn->fetchAssoc('SELECT * FROM world WHERE id = ?', [random_int(1, 10000)]);
         }
 
-        if ($queries == 1 && !$request->query->has('queries')) {
+        if ($queries === 1 && !$request->query->has('queries')) {
             $worlds = $worlds[0];
         }
 
         return new JsonResponse($worlds);
     }
 
-    public function updateAction(Request $request)
+    /**
+     * @Route("/update")
+     * @throws Exception
+     */
+    public function updateAction(Request $request): JsonResponse
     {
       $queries = $request->query->getInt('queries', 1);
       $queries = min(500, max(1, $queries));
 
-      $worlds = array();
+      $worlds = [];
       $em = $this->getDoctrine()->getManager();
       $repo = $this->getDoctrine()->getRepository(World::class);
 
       for ($i = 0; $i < $queries; ++$i) {
-        $world = $repo->find(mt_rand(1, 10000));
-        $randomNumber = mt_rand(1, 10000);
-        $world->setRandomNumber($randomNumber);
-        $worlds[] = $world;
+        $world = $repo->find(random_int(1, 10000));
+        if ($world) {
+            $randomNumber = random_int(1, 10000);
+            $world->setRandomNumber($randomNumber);
+            $worlds[] = $world;
+        }
       }
 
       $em->flush();
 
       return new JsonResponse($worlds);
     }
 
-    public function updateRawAction(Request $request)
+    /**
+     * @Route("/update-raw")
+     * @throws Exception
+     */
+    public function updateRawAction(Request $request): JsonResponse
     {
       $queries = $request->query->getInt('queries', 1);
       $queries = min(500, max(1, $queries));
 
-      $worlds = array();
+      $worlds = [];
       $conn = $this->get('database_connection');
 
       for ($i = 0; $i < $queries; ++$i) {
-          $id = mt_rand(1, 10000);
-          $randomNumber = mt_rand(1, 10000);
-          $conn->executeUpdate('UPDATE world SET randomNumber=? WHERE id=?', array($randomNumber, $id));
-          $worlds[] = array('id' => $id, 'randomNumber' => $randomNumber);
+          $id = random_int(1, 10000);
```

Comment:
as this is a performance test, using random_int as cryptographically secure function will make this code slower for sure.

Reply:
true

most of the other PHP frameworks are using the the same random so using any other would be unfair to use faster function (e.g. https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/PHP/laravel/app/Http/Controllers/Controller.php#L24)

Reply:
it's not only about speed but semantics. we don't need cryptographically secure integers here. so no point in using it. you could also argue it's not fair for php when compared to other languages.

Reply:
i agree, changed it back to `mt_rand`

Reply:
findings: while `random_int` is 30-40 times slower than `mt_rand` it has no significant effect on test performance (it can be theoretically called max 500 times, in practice this number is 20)